conans/client/tools/settings.py

```diff
@@ -0,0 +1,63 @@
+from conans.errors import ConanInvalidConfiguration
+from conans.client.build.cppstd_flags import cppstd_flag, cppstd_from_settings
+from conans.client.tools.oss import OSInfo
+
+
+def check_min_cppstd(conanfile, cppstd, gnu_extensions=False):
+    """ Check if current cppstd fits the minimal version required.
+
+        In case the current cppstd doesn't fit the minimal version required
+        by cppstd, a ConanInvalidConfiguration exception will be raised.
+
+    :param conanfile: ConanFile instance with cppstd to be compared
+    :param cppstd: Minimal cppstd version required
+    :param gnu_extensions: GNU extension is required (e.g gnu17). This option ONLY works on Linux.
+    """
+    assert (cppstd is not None), "Cannot check invalid cppstd version"
+    assert (conanfile is not None), "conanfile must be a ConanFile object"
+    assert (str(cppstd).isdigit()), "cppstd must be a number"
+
+    def less_than(lhs, rhs):
+        def extract_cpp_version(cppstd):
+            return str(cppstd).replace("gnu", "")
+
+        def add_millennium(cppstd):
+            return "19%s" % cppstd if cppstd == "98" else "20%s" % cppstd
+
+        lhs = add_millennium(extract_cpp_version(lhs))
+        rhs = add_millennium(extract_cpp_version(rhs))
+        return lhs < rhs
+
+    current_cppstd = cppstd_from_settings(conanfile.settings)
+    if current_cppstd and gnu_extensions and "gnu" not in current_cppstd and OSInfo().is_linux:
```

Comment:
The ``OSInfo().is_linux`` shouldn't be used, what if cross-building? The current OS cannot decide this logic, it seems that this should be a pure logic function from the input settings.

Reply:
I've added `os_build` first. But for header-only cases, where there is no settings, we need to check by OSInfo, because only gcc on linux has extensions.